### ISSUE TITLE
Problem disabling NFS export! aborting

### ISSUE THREAD

**Issue by erwis-ict** [brainupdaters/drlm#108]
Issue type: **Bug**
How often it happens: Always
Brief description of the issue:

Hello!

After server reboot when drlm starts there is an error:
"Problem disabling NFS export! aborting"
or
"exportfs: Could not find 'imis:/var/lib/drlm/store/imis' to unexport."

2019-03-03 10:30:02 Deactivating previous DR store for client: client01...
exportfs: Could not find 'client01:/var/lib/drlm/store/client01' to unexport.
2019-03-03 10:30:04 ERROR: drlm:runbackup:NFS:DISABLE:client01: Problem disabling NFS export! aborting ...
=== Stack trace ===
Trace 0: /usr/sbin/drlm:254 main
Trace 1: /usr/share/drlm/lib/runbackup-workflow.sh:86 WORKFLOW_runbackup
Trace 2: /usr/share/drlm/lib/framework-functions.sh:85 SourceStage
Trace 3: /usr/share/drlm/lib/framework-functions.sh:40 Source
Trace 4: /usr/share/drlm/backup/run/default/059_prep_backup.sh:19 source
Message: drlm:runbackup:NFS:DISABLE:client01: Problem disabling NFS export! aborting ...

exportfs
returns
/var/lib/drlm/store/client01
                <world>
				

after issuing command:
exportfs -vu *:/var/lib/drlm/store/client01
everything works again!

Replacing  "${CLI_NAME}" by "*" in function "disable_nfs_fs" in "/usr/share/drlm/lib/nfs_functions.sh" resolves the problem.

function disable_nfs_fs ()
{
  local CLI_NAME=$1
  
  if [[ $(exportfs | grep -w ${STORDIR}/${CLI_NAME}) ]]; then
    #exportfs -vu ${CLI_NAME}:${STORDIR}/${CLI_NAME}
    #changed to:
    exportfs -vu *:${STORDIR}/${CLI_NAME}
    if [ $? -eq 0 ]; then sleep 1; exportfs -f; return 0; else return 1; fi
    # Return 0 if OK or 1 if NOK
  else
    return 0
  fi
}

**Comment by proura** [comment]
Hello @erwis-ict,

What OS version in the DRLM server and client are you using? I will try to reproduce the bug, because if I use `exportfs -vu *:${STORDIR}/${CLI_NAME}` in my environment it cause and error `exportfs: Could not find '*:/var/lib/drlm/store/cilent1' to unexport.`

**Comment by erwis-ict** [comment]
Hello @proura,

For the DRLM server I'm using Centos 7 minimal install.
For the client Centos 6.

Best regards,

Eryk


Op 4-3-2019 om 8:32 schreef proura:

**Comment by proura** [comment]
Hello @erwis-ict,
![image](https://user-images.githubusercontent.com/5682570/53825322-4eaa8680-3f76-11e9-9e8d-cb95d13358e4.png)
I have tried to reproduce your bug without success, I can make a backup after restart DRLM server without problems and if I use the comand `exportfs -vu *:${STORDIR}/${CLI_NAME}` I get an error. What version of DRLM and ReaR are you using? In your reported error two different client names appear, client01 and imis, have you changed the hostname after install de client in the DRLM server? 

Best regards,

**Comment by erwis-ict** [comment]
Hello @proura,

I know where the problem comes from.
By mistake I have exported "/var/lib/drlm/store/client01" to the <world> 
by hand.
Then I rebooted the server and could not make backup anymore because of 
this error.
To get it work again I had to unexport it with a command "exportfs -ua".

Sorry for inconvenience and thank you for your time and prompt response.

Best regards,
erwis-ict

Op 5-3-2019 om 18:52 schreef proura:
